src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeMemory.Unix.cs

```diff
@@ -0,0 +1,126 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>This class contains methods that are mainly used to manage native memory.</summary>
+    public static unsafe class NativeMemory
+    {
+        /// <summary>Allocates an aligned block of memory of the specified size and alignment, in bytes.</summary>
+        /// <param name="alignment">The alignment, in bytes, of the block to allocate.</param>
+        /// <param name="byteCount">The size, in bytes, of the block to allocate.</param>
+        /// <returns>A pointer to the allocated aligned block of memory.</returns>
+        /// <exception cref="OutOfMemoryException">Allocating <paramref name="byteCount" /> of memory failed.</exception>
+        /// <remarks>
+        ///     <para>This method converts a <paramref name="byteCount" /> of <c>0</c> to <c>1</c>.</para>
+        ///     <para>This method is a thin wrapper over the C <c>aligned_alloc</c> API or a platform dependent aligned allocation API such as <c>_aligned_malloc</c> on Win32.</para>
+        /// </remarks>
+        [CLSCompliant(false)]
+        public static void* AlignedAlloc(nuint alignment, nuint byteCount)
```

Comment:
This argument ordering violates the usual .NET convention in appending the more specialized arguments last. Should `byteCount` be first - to align with the one and only argument for `Alloc` ?

Reply:
API review had discussed this and chose to mirror the C API here, rather than our own conventions.

Users are likely to get tripped up regardless of which order we do it, as they will either expect the C ordering or the Windows ordering or the .NET ordering.

Reply:
I'll bring this up when I raise the other items to API review.